autopilot/graph.go

```diff
@@ -136,6 +136,42 @@ func (d *databaseChannelGraph) ForEachNode(cb func(Node) error) error {
 	})
 }
 
+// FetchNode returns the Node from the graph for the given NodeID, or
+// ErrNodeNotFound if the node is not found. ErrNoAddresses is returned if the
+// node is found, but has no addresses to connected to.
+//
+// NOTE: Part of the autopilot.ChannelGraph interface.
+func (d *databaseChannelGraph) FetchNode(pubkeyBytes NodeID) (Node, error) {
```

Comment:
We may eventually also want to expose the feature bit information as well. As eventually the pilot may be aware of things like watch towers. As an example, if we don't have any channels yet, and want to open a channel, then it may make sense to open a channel up to a tower so we can pay it for tokens to watch over other channels. ofc, we'd need to ensure that we don't have it watch channels that we have directly with the node.